From: Thomas Klute <thomas2.klute@uni-dortmund.de>
Date: Thu, 22 Dec 2016 22:10:07 +0100
Subject: Test suite: Ensure CRLF line ends in HTTP headers

Debian Sid updated Apache from 2.4.23 to 2.4.25 yesterday, and 2.4.24
introduces stricter checks for the request syntax described in RFC
7230. This breaks test cases where the request input file did not
contain CRLF line ends. Fix by forcing CRLF line ends when sending
requests in the runtest script.
---
 test/runtests | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/test/runtests b/test/runtests
index 5c8afe8..a253686 100755
--- a/test/runtests
+++ b/test/runtests
@@ -207,7 +207,11 @@ sleep_pidfile="$(mktemp mod_gnutls_test-XXXXXX.pid)"
 # failing tests. Sending sleep to the background allows the test
 # case to proceed instead of waiting for it to return. The sleep
 # process is stopped after gnutls-cli terminates.
-if (sed "s/__HOSTNAME__/${TEST_HOST}/" <${t}/input && \
+#
+# The line end manipulation in sed guarantees that all header lines
+# end with CRLF as required by RFC 7230, Section 3.1.1 regardless of
+# the line ends in the input file.
+if (sed -r "s/__HOSTNAME__/${TEST_HOST}/;s/\r?$/\r/" <${t}/input && \
 	   run_with_pidfile "${sleep_pidfile}" sleep "${TEST_QUERY_DELAY}" &) | \
        gnutls-cli -p "${TEST_PORT}" $(cat ${t}/gnutls-cli.args) "${TEST_HOST}" \
        | tee "$output" && test "${PIPESTATUS[1]}" -eq 0;
